crypto/groups/qr_special_rsa.go

```diff
@@ -41,29 +42,46 @@ func NewQRSpecialRSA(safePrimeBitLength int) (*QRSpecialRSA, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	qrRSA, err := NewQRRSA(specialRSAPrimes.P, specialRSAPrimes.Q)
+	if err != nil {
+		return nil, err
+	}
 	return &QRSpecialRSA{
-		N:      new(big.Int).Mul(specialRSAPrimes.P, specialRSAPrimes.Q),
+		N:      qrRSA.N,
 		P:      specialRSAPrimes.P,
 		Q:      specialRSAPrimes.Q,
 		SmallP: specialRSAPrimes.P1,
 		SmallQ: specialRSAPrimes.Q1,
```

Comment:
I'm sorry, I should have made this comment in your previous PR, but it would be much clearer if the fields in `type SpecialRSAPrimes struct` were named `smallP` and `smallQ` instead of `P1` and `Q1`.

Reply:
I agree, this was a mess. I changed everywhere to P1 and Q1.